### PR TITLE
CSHARP-5288: Make types internal in Operations namespace

### PR DESCRIPTION
**Pull request by BorisDog**

### CODE REVIEW

File: src/MongoDB.Driver/Core/Operations/CreateIndexRequest.cs [hunk 1]
```diff
@@ -19,359 +19,92 @@
...
-        /// Gets the name of the index.
-        /// </summary>
-        /// <returns>The name of the index.</returns>
+        public double? BucketSize { get; set; }
+        public Collation Collation { get; set; }
+        public string DefaultLanguage { get; set; }
+        public TimeSpan? ExpireAfter { get; set; }
+        public bool? Hidden { get; set; }
+        public string LanguageOverride { get; set; }
+        public BsonDocument Keys { get; set; }
```
Comment (sanych-sun) [inline src/MongoDB.Driver/Core/Operations/CreateIndexRequest.cs:41]: `Keys` use to be readonly property.

File: src/MongoDB.Driver/Core/Operations/CreateSearchIndexesOperation.cs [hunk 2]
```diff
@@ -34,7 +34,8 @@ internal sealed class CreateSearchIndexesOperation : IWriteOperation<BsonDocumen
         // fields
         private readonly CollectionNamespace _collectionNamespace;
         private readonly MessageEncoderSettings _messageEncoderSettings;
-        private readonly IEnumerable<CreateSearchIndexRequest> _requests;
+        private readonly IEnumerable<
+            CreateSearchIndexRequest> _requests;
```
Comment (sanych-sun) [inline src/MongoDB.Driver/Core/Operations/CreateSearchIndexesOperation.cs:38]: Unintentional new line? Please revert.

File: src/MongoDB.Driver/Core/Operations/DeleteRequest.cs [hunk 3]
```diff
@@ -19,76 +19,23 @@
...
-        /// </remarks>
-        /// <value>
-        /// A limit on the number of documents that should be deleted.
-        /// </value>
-        public int Limit
-        {
-            get { return _limit; }
-            set { _limit = value; }
-        }
+        public Collation Collation { get; set; }
```
Comment (sanych-sun) [inline src/MongoDB.Driver/Core/Operations/DeleteRequest.cs:33]: Should all properties be init only?

File: src/MongoDB.Driver/Core/Operations/UpdateRequest.cs [hunk 4]
```diff
@@ -14,131 +14,39 @@
...
-        }
-
-        /// <summary>
-        /// Gets the update type.
-        /// </summary>
-        public UpdateType UpdateType
-        {
-            get { return _updateType; }
-        }
+        public IEnumerable<BsonDocument> ArrayFilters { get; set; }
```
Comment (sanych-sun) [inline src/MongoDB.Driver/Core/Operations/UpdateRequest.cs:36]: Init only properties?

File: src/MongoDB.Driver/Core/Operations/WriteRequest.cs [hunk 5]
```diff
@@ -18,55 +18,20 @@
...
-        /// Gets the request type.
-        /// </summary>
-        /// <value>
-        /// The request type.
-        /// </value>
-        public WriteRequestType RequestType
-        {
-            get { return _requestType; }
-        }
+        public int? CorrelationId { get; set; }
```
Comment (sanych-sun) [inline src/MongoDB.Driver/Core/Operations/WriteRequest.cs:31]: Init only?

**Review by sanych-sun — APPROVED** [review]
LGTM + minor suggestions.

File: src/MongoDB.Driver/Core/Operations/UpdateRequest.cs (hunk 4 above)
Comment (BorisDog) [reply src/MongoDB.Driver/Core/Operations/UpdateRequest.cs]: Done.

File: src/MongoDB.Driver/Core/Operations/CreateSearchIndexesOperation.cs (hunk 2 above)
Comment (BorisDog) [reply src/MongoDB.Driver/Core/Operations/CreateSearchIndexesOperation.cs]: Thanks, fixed.

File: src/MongoDB.Driver/Core/Operations/CreateIndexRequest.cs (hunk 1 above)
Comment (BorisDog) [reply src/MongoDB.Driver/Core/Operations/CreateIndexRequest.cs]: Good catch, done.

**Pull request merged**